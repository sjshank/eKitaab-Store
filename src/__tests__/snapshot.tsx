import { render } from "@testing-library/react";
import Home from "../pages/index";
import { it } from "node:test";
import { expect } from "@jest/globals";

const mockDashboard = [
  {
    label: "Books Listed",
    count: 10,
    muiIcon: "LibraryBooksRounded",
  },
  {
    label: "Authors Registered",
    count: 20,
    muiIcon: "SupervisedUserCircleRounded",
  },
  {
    label: "Genres Available",
    count: 30,
    muiIcon: "CategoryRounded",
  },
  {
    label: "Book Instances",
    count: 40,
    muiIcon: "ContentCopyRounded",
  },
  {
    label: "Copies Available",
    count: 50,
    muiIcon: "EventAvailableRounded",
  },
];

it("renders homepage unchanged", () => {
  const { container } = render(<Home dashboard={mockDashboard} />);
  expect(container).toMatchSnapshot();
});
