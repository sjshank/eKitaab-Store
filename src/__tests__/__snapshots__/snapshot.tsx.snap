// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <section
    class="MuiBox-root css-0"
  >
    <h2
      class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Library Dashboard
    </h2>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1ndtn4l-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1ndtn4l-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1ndtn4l-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1ndtn4l-MuiSkeleton-root"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1ndtn4l-MuiSkeleton-root"
      />
    </div>
    <div
      class="MuiBox-root css-1lwe5qo"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
        href="/catalog/books"
        role="link"
        tabindex="0"
        title="Explore Library Catalog"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yr159y-MuiSvgIcon-root"
          data-testid="TravelExploreRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19.3 16.9c.58-1.01.95-2.23.51-3.65-.53-1.72-2.04-3.05-3.84-3.22-2.87-.28-5.23 2.07-4.95 4.95.18 1.79 1.5 3.31 3.22 3.84 1.43.44 2.64.07 3.65-.51l2.5 2.5c.39.39 1.01.39 1.4 0 .39-.39.39-1.01 0-1.4zm-3.8.1c-1.4 0-2.5-1.1-2.5-2.5s1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5-1.1 2.5-2.5 2.5M12 20v2C6.48 22 2 17.52 2 12S6.48 2 12 2c4.84 0 8.87 3.44 9.8 8h-2.07c-.64-2.46-2.4-4.47-4.73-5.41V5c0 1.1-.9 2-2 2h-2v2c0 .55-.45 1-1 1H8v2h2v3H9l-4.79-4.79C4.08 10.79 4 11.38 4 12c0 4.41 3.59 8 8 8"
          />
        </svg>
        <span
          class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          Explore Library Catalog
        </span>
      </a>
    </div>
  </section>
</div>
`;
